### DISCUSSION TITLE
After the window is enlarged, the window inside will change to the same as before switching

### DISCUSSION THREAD

**Discussion by coolH-alt** [fltk-rs/fltk-rs#1198] [Q&A]
Enlarge the window and set the size of other windows
After zooming out, it is normal. Set the size of other windows again. At this time, zooming in the window will become the same as when switching for the first time

let mut vedio_win = Window::new(
        0,
        title_flex.h(),
        left_win.w(),
        media_container.h() - title_flex.h(),
        None,
    );

    let mut vedio_win_ch1 = Window::new(0, 0, vedio_win.w(), vedio_win.h(), None);

    vedio_win_ch1.make_resizable(true);
    vedio_win_ch1.set_color(Color::Black);
    vedio_win_ch1.end();

    let mut vedio_win_ch2 = Window::new(
        vedio_win.w() / 2 + v_margin,
        0,
        vedio_win.w() / 2 - v_margin,
        vedio_win.h() / 2 - v_margin,
        None,
    );
    vedio_win_ch2.set_color(Color::Black);
    vedio_win_ch2.end();

    let mut vedio_win_ch3 = Window::new(
        vedio_win.w() / 2 + v_margin,
        vedio_win.h() / 2 + v_margin,
        vedio_win.w() / 2 - v_margin,
        vedio_win.h() / 2 - v_margin,
        None,
    );
    vedio_win_ch3.set_color(Color::Black);
    vedio_win_ch3.end();

    let mut vedio_win_ch4 = Window::new(
        0,
        vedio_win.h() / 2 + v_margin,
        vedio_win.w() / 2 - v_margin,
        vedio_win.h() / 2 - v_margin,
        None,
    );
    vedio_win_ch4.set_color(Color::Black);
    vedio_win_ch4.end();

    vedio_win.resizable(&vedio_win_ch1);
    vedio_win.resizable(&vedio_win_ch2);
    vedio_win.resizable(&vedio_win_ch3);
    vedio_win.resizable(&vedio_win_ch4);
    vedio_win.make_resizable(true);
    vedio_win.set_color(Color::White);
    vedio_win.end();

Message::Split(msg) => match msg {
                    SplitMessage::F1 => {
                        vedio_win_ch1.resize(0, 0, vedio_win.w(), vedio_win.h());
                        screen_num = 1;
                        vedio_win_ch1.redraw();
                        vedio_win_ch2.redraw();
                        vedio_win_ch3.redraw();
                        vedio_win_ch4.redraw();
                    }
                    SplitMessage::F2 => {
                        vedio_win_ch1.resize(0, 0, vedio_win.w() / 2 - v_margin, vedio_win.h());
                        vedio_win_ch2.resize(
                            vedio_win.w() / 2 + v_margin,
                            0,
                            vedio_win.w() / 2 - v_margin,
                            vedio_win.h(),
                        );
                        screen_num = 2;
                        vedio_win_ch1.redraw();
                        vedio_win_ch2.redraw();
                        vedio_win_ch3.redraw();
                        vedio_win_ch4.redraw();
                    }
                    SplitMessage::F3 => {
                        vedio_win_ch1.resize(0, 0, vedio_win.w() / 2 - v_margin, vedio_win.h());
                        vedio_win_ch2.resize(
                            vedio_win.w() / 2 + v_margin,
                            0,
                            vedio_win.w() / 2 - v_margin,
                            vedio_win.h() / 2 - v_margin,
                        );

                        vedio_win_ch3.resize(
                            vedio_win.w() / 2 + v_margin,
                            vedio_win.h() / 2 + v_margin,
                            vedio_win.w() / 2 - v_margin,
                            vedio_win.h() / 2 - v_margin,
                        );
                        screen_num = 3;
                        vedio_win_ch1.redraw();
                        vedio_win_ch2.redraw();
                        vedio_win_ch3.redraw();
                        vedio_win_ch4.redraw();
                    }
                    SplitMessage::F4 => {
                        vedio_win_ch1.resize(
                            0,
                            0,
                            vedio_win.w() / 2 - v_margin,
                            vedio_win.h() / 2 - v_margin,
                        );
                        vedio_win_ch2.resize(
                            vedio_win.w() / 2 + v_margin,
                            0,
                            vedio_win.w() / 2 - v_margin,
                            vedio_win.h() / 2 - v_margin,
                        );

                        vedio_win_ch3.resize(
                            vedio_win.w() / 2 + v_margin,
                            vedio_win.h() / 2 + v_margin,
                            vedio_win.w() / 2 - v_margin,
                            vedio_win.h() / 2 - v_margin,
                        );

                        vedio_win_ch4.resize(
                            0,
                            vedio_win.h() / 2 + v_margin,
                            vedio_win.w() / 2 - v_margin,
                            vedio_win.h() / 2 - v_margin,
                        );
                        screen_num = 4;
                        vedio_win_ch1.redraw();
                        vedio_win_ch2.redraw();
                        vedio_win_ch3.redraw();
                        vedio_win_ch4.redraw();
                    }
                },

**Comment by MoAlyousef** [comment]
Not all platforms support overlapping child windows. I've tried this code on linux and windows:
```rust
use fltk::{enums::*, prelude::*, *};

fn main() {
    let a = app::App::default();
    let mut screen_num = 0;
    let v_margin = 5;
    let mut vedio_win = window::Window::new(0, 0, 800, 600, None);

    let mut vedio_win_ch1 = window::Window::new(0, 0, vedio_win.w() / 2 - v_margin, vedio_win.h() / 2 - v_margin, None);

    vedio_win_ch1.set_color(Color::Green);
    vedio_win_ch1.end();

    let mut vedio_win_ch2 = window::Window::new(
        vedio_win.w() / 2 + v_margin,
        0,
        vedio_win.w() / 2 - v_margin,
        vedio_win.h() / 2 - v_margin,
        None,
    );
    vedio_win_ch2.set_color(Color::Red);
    vedio_win_ch2.end();

    let mut vedio_win_ch3 = window::Window::new(
        vedio_win.w() / 2 + v_margin,
        vedio_win.h() / 2 + v_margin,
        vedio_win.w() / 2 - v_margin,
        vedio_win.h() / 2 - v_margin,
        None,
    );
    vedio_win_ch3.set_color(Color::Cyan);
    vedio_win_ch3.end();

    let mut vedio_win_ch4 = window::Window::new(
        0,
        vedio_win.h() / 2 + v_margin,
        vedio_win.w() / 2 - v_margin,
        vedio_win.h() / 2 - v_margin,
        None,
    );
    vedio_win_ch4.set_color(Color::Light1);
    vedio_win_ch4.end();

    vedio_win.make_resizable(true);
    vedio_win.set_color(Color::White);
    vedio_win.end();
    vedio_win.show();

    a.run().unwrap();
}
```
And it seems to be working correctly. Your code seems to give vedio_win_ch1 the same size as its parent, which interferes with other windows children of vedio_win. Giving it the same size as the others seems to work fine.
